zanata-war/src/main/java/org/zanata/service/impl/TranslationServiceImpl.java

```diff
@@ -859,31 +878,36 @@ public String apply(TextFlowTarget input) {
                     hTarget.setVersionNum(hTarget.getVersionNum() + 1);
 
                     changed = true;
-                    Long actorId;
                     if (assignCreditToUploader){
                         HPerson hPerson = authenticatedAccount.getPerson();
                         hTarget.setTranslator(hPerson);
                         hTarget.setLastModifiedBy(hPerson);
-                        actorId = hPerson.getId();
                     } else {
                         hTarget.setTranslator(null);
                         hTarget.setLastModifiedBy(authenticatedAccount.getPerson());
-                        actorId = null;
                     }
                     hTarget.setSourceType(translationSourceType);
                     hTarget.setCopiedEntityId(null);
                     hTarget.setCopiedEntityId(null);
                     textFlowTargetDAO.makePersistent(hTarget);
-                    signalPostTranslateEvent(actorId, hTarget, currentState);
+                    updateTextFlowTargetStateEvent(eventMap, hTarget, currentState);
                 }
             }
-
             if (handleOp.isPresent()) {
                 handleOp.get().increaseProgress(1);
             }
         }
+        Long actorId =
+            assignCreditToUploader ? authenticatedAccount.getPerson().getId() :
+                null;
+        for (Map.Entry<DocumentLocaleKey, Collection<TextFlowTargetState>> entry : eventMap
+            .asMap().entrySet()) {
```

Comment:
This method (saveBatch) works on a single HLocale and HDocument, so there should only be a single TextFlowTargetStateEvent. We shouldn't need the Map/MultiMap here either.